### DISCUSSION TITLE
How to get parent collection data for building sub collection

### DISCUSSION THREAD

**Discussion by LuLue7775** [firecmsco/firecms#491] [Q&A] [answered]
Hi, having trouble passing data among collections
Can I get parent data in callbacks?

`const subCollection = buildCollection<Courses>({
    path: "courses",
    name: "Courses",
    callbacks: coursesCallbacks,
    properties: {
        name: {
            name: "name",
            dataType: "string"
        },
    }

})
`

Not sure if I have it right. 
Please I've been stuck for days..

**Comment by LuLue7775** [comment]
Also, is there a way to get data from another collection in buildCollection? just like useDataSource but in buildCollection

**Reply by fgatti675** [reply, marked as answer]
Yes, you can use a collection builder
https://firecms.co/docs/navigation/
you get the datasource directly in the callback

**Reply by LuLue7775** [reply]
Thank you! 
But I can't access specific path in delete callback, is there anyway to do it? 

`
  onDelete: ({
        entity,
        path,
        context
    }: EntityOnDeleteProps<Course>) => {
        context.dataSource.deleteEntity({
            entity: entity,
        })
    },

`
